### ISSUE TITLE
Expectation values of FermionOperator with respect to fermionic Gaussian state

### ISSUE THREAD

**Issue by michaelkaicher** [quantumlib/OpenFermion#592]
Hi all, 

this is probably already implemented, (it's probably more a pull request than an issue) but I lately wrote some code that computes expectation values of a (normal-ordered) FermionOperator provided by OpenFermion with respect to a fermionic Gaussian state, more precisely its covariance matrix, that outputs the value using Wick's theorem. I was wondering whether there would be code in Open Fermion that I could use to compare it to (runtime, if it produces the same number, etc...). 

To summarize, is there a function that does as follows: 
Argument: FermionOperator, covariance matrix, number of spin-orbitals
Returns: Expectation value of FermionOperator w.r.t. the fermionic Gaussian state described by the covariance matrix

Cheers, 
Michael.

**Comment by ncrubin** [comment]
Hi Michael.  To my knowledge we don't have something explicitly for this.  The closest thing we have is wedge product which can produce arbitrary RDMs from a particle conserving covariance matrix.  It would be great to have your contribution as we push to expand this area.  There are been an increasing number of proposal for using fermionic gaussian states as benchmarking targets.  Feel free to open a PR!

**Comment by michaelkaicher** [comment]
Thanks! I will try to do this as soon as I find time!
